gfx.h

```diff
@@ -1402,6 +1402,24 @@ class GfxInternal
         device_->QueryInterface(IID_PPV_ARGS(&mesh_device_));
         SetDebugName(device_, "gfx_Device");
 
+        if((flags & kGfxCreateContextFlag_EnableDebugLayer) != 0)
+        {
+            ID3D12InfoQueue1 *debugCallback;
+            device_->QueryInterface(IID_PPV_ARGS(&debugCallback));
+            DWORD cookie;
+            D3D12MessageFunc callback = [](D3D12_MESSAGE_CATEGORY Category, D3D12_MESSAGE_SEVERITY Severity,
+                                            D3D12_MESSAGE_ID ID, LPCSTR pDescription, void *pContext) {
+                if(Severity <= D3D12_MESSAGE_SEVERITY_ERROR)
+                {
+                    GFX_PRINTLN("D3D12 Error: %s", pDescription);
+                    GFX_ASSERT(false);
```

Comment:
You could use `GFX_ASSERTMSG(0, "D3D12 Error: %s", pDescription);`, which would do the same thing but in a single line.

Reply:
I actually left it as 2 lines so that the debug message was already printed by the time the assert was triggered. The messages are rather long so its quicker and easier to view them once printed

Reply:
`GFX_ASSERTMSG()` does print out the message prior to the breakpoint, so that should be the same behaviour.

Reply:
fair enough, changed